account_purchase_stock_report_non_billed/models/stock_move.py

```diff
@@ -60,6 +60,15 @@ def _set_not_invoiced_values(self, qty_to_invoice, invoiced_qty):
             self.price_not_invoiced = (
                 qty_to_invoice - invoiced_qty
             ) * self.purchase_line_id.price_unit
+            if "discount" in self.purchase_line_id._fields:
+                self.price_not_invoiced = (qty_to_invoice - invoiced_qty) * (
+                    self.purchase_line_id.price_unit
+                    - (
+                        self.purchase_line_id.discount
+                        * self.purchase_line_id.price_unit
+                        / 100
+                    )
+                )
```

Comment:
Formula is not correct and better to not repeat operations:

```suggestion
            price_unit = self.purchase_line_id.price_unit
            if "discount" in self.purchase_line_id._fields:
                price_unit = self.purchase_line_id.price_unit * (1 - self.purchase_line_id.discount / 100)
            self.price_not_invoiced = (
                qty_to_invoice - invoiced_qty
            ) * price_unit
```